.github/workflows/main.yml

```diff
@@ -1,42 +1,188 @@
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+          
+      - name: Run ineffassign
+        run: |
+          cd /tmp
+          go install github.com/gordonklaus/ineffassign@latest
+          cd ${{ env.GOPATH }}/src/github.com/G-Research/armada
+          echo "Run ineffassign..."
+          ineffassign ./...
+
+      - name: Check formatting
+        run: |
+          go install golang.org/x/tools/cmd/goimports@v0.1.1
+          if [ $(goimports -l -local "github.com/G-Research/armada" | wc -l) -eq 0 ]; then exit 0; else exit 1; fi
+
+      - name: Check generated files
+        run: |
+          make proto
+          git status -s -uno
+          git --no-pager diff
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+  test:
+    name: Test
+    needs: [code_styles]
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg/mod
+            ~/.cache/go-build
+            /home/runner/.cache/go-build
+            ~/Library/Caches/go-build
+            %LocalAppData%\go-build
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Run tests
+        run: |
+          make tests
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Check go.sum
+        run: |
+          cat go.sum
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+
+  build:
+    name: Build
+    needs: [test] 
+    runs-on: ubuntu-20.04
+    env: #TODO change DOCKERTOKEN and DOCKERUSERNAME
+      dockerToken: ${{ secrets.DOCKERTOKEN }}
+      dockerUsername: ${{ secrets.DOCKERUSERNAME }}
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - name: Checkout code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Start cluster for e2e tests
+        run: |
+          ./e2e/setup/setup_cluster_ci.sh
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Build & Run e2e tests
+        run: | 
+          make build-ci tests-e2e 
+        working-directory: ${{ env.GOPATH }}/src/github.com/G-Research/armada
+
+      - name: Check go.sum

```

Comment:
what is the purpose of this step, do we only want to print the sum? if yes I would potentially change this to "print go.sum"
_same for line 95_

Reply:
no idea really 😨 , just copied the same behavior from [CircleCI](https://github.com/G-Research/armada/blob/master/.circleci/config.yml#L165)
as I'm unaware of team process and habits, i coppied it over 🤸‍♂️